api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/ehcache/Ehcache3Properties.java

```diff
@@ -0,0 +1,163 @@
+package org.apereo.cas.configuration.model.support.ehcache;
+
+import org.apereo.cas.configuration.model.core.util.EncryptionRandomizedSigningJwtCryptographyProperties;
+import org.apereo.cas.configuration.support.RequiredProperty;
+import org.apereo.cas.configuration.support.RequiresModule;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.NestedConfigurationProperty;
+
+import java.io.Serializable;
+
+/**
+ * This is {@link Ehcache3Properties}.
+ *
+ * @author Hal Deadman
+ * @since 6.2.0
+ */
+@RequiresModule(name = "cas-server-support-ehcache3-ticket-registry")
+@Getter
+@Setter
+public class Ehcache3Properties implements Serializable {
+
+    private static final long serialVersionUID = 7772510035918976450L;
+
+    /**
+     * Enabled allows this registry to be disabled on startup (so registry choice can be made at runtime).
+     */
+    @RequiredProperty
+    private boolean enabled = true;
+
+    /**
+     * The name of the cache manager instance.
+     */
+    @RequiredProperty
+    private String cacheManagerName = "ticketRegistryCacheManager";
+
+    /**
+     * Builder that sets the maximum objects to be held in memory (0 = no limit).
+     */
+    private int maxElementsInMemory = 10_000;
+
+    /**
+     * Size of disk cache.
+     */
+    private int maxSizeOnDisk = 200;
+
+    /**
+     * Disk cache size units as defined by EhCache. (e.g. MB, GB)
+     */
+    private String maxSizeOnDiskUnits = "MB";
+
+    /**
+     * Per cache size of disk cache.
+     */
+    private int perCacheSizeOnDisk = 20;
+
+    /**
+     * Per cache disk cache size units as defined by EhCache. (e.g. MB, GB)
+     */
+    private String perCacheSizeOnDiskUnits = "MB";
```

Comment:
Not an action item, but more of a note for later; things like this can later be refactored into a common type similar to `Duration`, where you'd combine the unit of measure and value into one single setting with a type that can capture and parse all.